utils/auth-utils.ts

```diff
@@ -0,0 +1,9 @@
+import { Session, useSession as nextAuthUseSession } from "next-auth/client";
+import { SessionUser } from "&server/models/SessionUser";
+
+export type NPPSession = Session & { user: SessionUser };
+
+export function useSession(): [NPPSession, boolean] {
+  const [session, loading] = nextAuthUseSession();
+  return [session as NPPSession, loading];
+}
```

Comment:
Not sure if this will be useful, but it will give the correct types :man_shrugging: